Event migration improvements (#29514)
akka/akka: akka-docs/src/main/paradox/serialization-jackson.md

### Add Field

> #### Add Optional Field

Reviewer: maybe too many levels. would be enough with `### Add Optional Field` and `### Add Mandatory Field`

Author: There's some link to `#add-field` from another page and I just preferred keeping the title.

It's a list of class names or prefixes of class names.

> ## Rolling updates

Reviewer 2: ```suggestion
## Rolling upgrade
```

Since we are talking about updating a version, I think **upgrade** is a better term. Like in upgrade/downgrade.

Reviewer: I'm sure both are used rather interchangeably, but I prefer update since it's not always better (could be introducing new bugs).

External data point https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/

When doing a rolling update, for a period of time there are two different binaries running in production. If the schema
has evolved requiring a new schema version, the data serialized by the new binary will be unreadable from the old 
binary. This situation causes transient errors on the processes running the old binary. This service degradation is 

> usually fine since the rolling update will eventually complete and all old processes will be replaced with the new 

Reviewer 2: ```suggestion
usually fine since the rolling upgrade will eventually complete and all old processes will be replaced with the new 
```

binary. To avoid this service degradation you can also use forward-one support in your schema evolutions.

> To complete a no-degradation rolling update, you need to make two deloyments. First, deploy a new binary which can read 

Reviewer 2: ```suggestion
To complete a no-degradation rolling upgrade, you need to make two deployments. First, deploy a new binary which can read 
```

It's a list of class names or prefixes of class names.

## Rolling upgrades

> When doing a rolling upgrade, for a period of time there are two different binaries running in production. If the schema

Reviewer: I prefer "rolling update", but we have use some mix of both anyway

Author: raised https://github.com/akka/akka/pull/29523
